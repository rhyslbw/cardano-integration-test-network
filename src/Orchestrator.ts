import { createCardanoCli } from './CardanoCli'
import { createCardanoNodeCli } from './cardano-node/CardanoNodeCli'
import { Environment } from './Environment'

export interface BinaryVersions {
  cardanoCli: string,
  cardanoNode: string
}

export interface Orchestrator {
  binaryVersions: () => Promise<BinaryVersions>
}

export function createOrchestrator (environment: Environment): Orchestrator {
  const cardanoCli = createCardanoCli(environment.CARDANO_CLI_PATH)
  return {
    binaryVersions: async () => {
      return {
        cardanoCli: await cardanoCli.getVersion(),
        cardanoNode: await createCardanoNodeCli(environment.CARDANO_NODE_PATH).getVersion()
      }
    }
  }
}
