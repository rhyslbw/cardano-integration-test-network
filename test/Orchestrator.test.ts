import path from 'path'
import { createOrchestrator, Orchestrator } from '@src/Orchestrator'

describe('Orchestrator', () => {
  let orchestrator: Orchestrator

  beforeAll(() => {
    orchestrator = createOrchestrator({
      CARDANO_CLI_PATH: process.env.CARDANO_CLI_PATH || path.resolve(__dirname, '..', 'bin', 'cardano-cli'),
      CARDANO_NODE_PATH: process.env.CARDANO_NODE_PATH || path.resolve(__dirname, '..', 'bin', 'cardano-node')
    })
  })

  describe('binaryVersions', () => {
    it('returns the cardano software versions', async () => {
      const versions = await orchestrator.binaryVersions()
      expect(versions).toMatchSnapshot()
    })
  })
})
