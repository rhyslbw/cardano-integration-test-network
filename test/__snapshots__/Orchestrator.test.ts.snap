// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orchestrator binaryVersions returns the cardano software versions 1`] = `
Object {
  "cardanoCli": "cardano-cli 1.24.2 - linux-x86_64 - ghc-8.10
git rev 0000000000000000000000000000000000000000",
  "cardanoNode": "cardano-node 1.24.2 - linux-x86_64 - ghc-8.10
git rev 0000000000000000000000000000000000000000",
}
`;
